src/OrchardCore.Modules/OrchardCore.Users/Startup.cs

```diff
@@ -230,12 +230,16 @@ public override void ConfigureServices(IServiceCollection services)
     }
 
     [RequireFeatures("OrchardCore.Roles")]
-    public class RoleStartup : StartupBase
+    public class RolesStartup : StartupBase
     {
         public override void ConfigureServices(IServiceCollection services)
         {
-            services.AddScoped<IAuthorizationHandler, UserAuthorizationHandler>();
+            services.AddScoped<IRoleRemovedEventHandler, UserRoleRemovedEventHandler>();
+            services.AddSingleton<IIndexProvider, UserByRoleNameIndexProvider>();
```

Comment:
I actually removed it from `AddUsers()` instead. Its is a role index, since it is a role related I think it should be part of the role feature not the user.